src/lib/packager/clients/software_proposal.rb

```diff
@@ -0,0 +1,169 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "installation/proposal_client"
+
+module Yast
+  class SoftwareProposalClient < ::Installation::ProposalClient
+    def initialize
+      Yast.import "Pkg"
+      textdomain "packager"
+
+      Yast.import "Packages"
+      Yast.import "Language"
+      Yast.import "Installation"
+    end
+
+    def make_proposal(flags)
+      @force_reset = flags.fetch("force_reset", false)
+
+      @language_changed = adjust_locales
+      @language_changed ||= flags.fetch("language_changed", false)
```

Comment:
well, it is usually discouraged to use `||=` for booleans, as it is not easy to recognize if it is mistake or intention.